sql/src/main/java/io/crate/executor/PagableTaskResult.java

```diff
@@ -21,13 +21,42 @@
 
 package io.crate.executor;
 
+import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
 
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+
 /**
  * enhanced TaskResult that supports fetching the next page
  * asynchronously.
  */
-public interface PagableTaskResult extends TaskResult {
+public interface PagableTaskResult extends TaskResult, Closeable {
+
+    public static final PagableTaskResult EMPTY_PAGABLE_RESULT = new PagableTaskResult() {
+        @Override
+        public ListenableFuture<PagableTaskResult> fetch(PageInfo pageInfo) {
+            return Futures.immediateFailedFuture(new NoSuchElementException());
+        }
+
+        @Override
+        public Object[][] rows() {
+            return TaskResult.EMPTY_ROWS;
+        }
+
+        @Nullable
+        @Override
+        public String errorMessage() {
+            return null;
+        }
+
+        @Override
+        public void close() throws IOException {
```

Comment:
should probably have some javadocs in the interface to explain what classes implementing the interface are expected to do in the functions?